src/interface/widgets/mainWindow.cpp

```diff
@@ -142,8 +149,15 @@ MainWindow::MainWindow(QWidget* parent) : QMainWindow(parent) {
     connect(robotsTimer, SIGNAL(timeout()), this, SLOT(updateRobotsWidget())); // we need to pass the visualizer so thats why a seperate function is used
     connect(robotsTimer, SIGNAL(timeout()), mainControlsWidget, SLOT(updatePause()));
     connect(robotsTimer, SIGNAL(timeout()), mainControlsWidget, SLOT(updateContents()));
+    robotsTimer->start(200); // 5fps
+
+    auto * graphTimer = new QTimer(this);
+    connect(graphTimer, SIGNAL(timeout()), graphWidget, SLOT(updateContents()));
+    graphTimer->start(200); // 5fps
```

Comment:
You could for updating also trigger updates when a new FPS data is set or when the internal clock is ticking, that'd be a bit more clean I think. This is fine though.